test/functional/services/esql.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// import expect from '@kbn/expect';
+import { FtrService } from '../ftr_provider_context';
+
+export class ESQLService extends FtrService {
+  private readonly retry = this.ctx.getService('retry');
+  private readonly testSubjects = this.ctx.getService('testSubjects');
+
+  public async getHistoryItems(): Promise<string[][]> {
+    const queryHistory = await this.testSubjects.find('TextBasedLangEditor-queryHistory');
+    const tableBody = await this.retry.try(async () => queryHistory.findByTagName('tbody'));
+    const $ = await tableBody.parseDomContent();
```

Comment:
Nice! I didn't know about this feature! I'm sure it's much more performant to load everything in one go.